File: Nodejs/services/services.js
var merge = require("../merge/merge.js")
,	bitcode = require("../bitcode/bitcode.js")
,	graph = require("../graph/kruskal.js");

exports.SortEmails = function(filename,cb) {
	let name = filename.split(".")[0];
	merge.FilterFile(name+"/"+filename,"(\\w[-._\\w]*\\w@\\w[-._\\w]*\\w\\.\\w{2,3})",function(err){
		if (err) {
			console.log("error filter");
		}else{
			console.log("termino filter");
			merge.CreateLeaves(name+"/"+filename+".filtered",5, function(err){
				if(err)
					console.log("error create leaves");
				else{
						console.log("termino filter");
						merge.MergeSort(name+"/leaves/",function(err){
							if (err) {
								console.log("error merge");
							}else{
								console.log("done: ",filename);
								cb(merge.GetSortedFile(filename));
							}
						});
					}
			});
		}
	});
};

exports.HideMessage = function(filename, message,cb) {
	bitcode.HideMessage(filename,message)
	cb(bitcode.GetHidden(filename))
}

exports.SeekMessage = function(filename,cb) {
	bitcode.SeekMessage(filename)
	cb(bitcode.GetMessage(filename))
}

exports.Kruskal = function(json_graph,cb) {
	cb(graph.ApplyKruskal(json_graph))
}

File: Python/services.py
import merge, bitcode, kruskal as graph

def SortEmails(filename):
	name = filename.split(".")[0]
	merge.FilterFile(name+"/"+filename,"(\\w[-._\\w]*\\w@\\w[-._\\w]*\\w\\.\\w{2,3})")
	merge.CreateLeaves(name+"/"+filename+".filtered",5)
	merge.MergeSort(name+"/leaves/")

	return merge.GetSortedFile(filename)

def HideMessage(filename, message):
	bitcode.HideMessage(filename,message)
	return bitcode.GetHidden(filename)

def SeekMessage(filename):
	bitcode.SeekMessage(filename)
	return bitcode.GetMessage(filename)

def Kruskal(json_graph):
	tree = graph.ApplyKruskal(json_graph)
	return str(tree)